pkg/controller/pxcbackup/controller.go

```diff
@@ -354,34 +352,40 @@ func (r *ReconcilePerconaXtraDBClusterBackup) runS3BackupFinalizer(cr *api.Perco
 
 	cr.SetFinalizers(finalizers)
 
-	logger.Info("backup was removed from s3", "name", cr.Name)
+	if err != nil {
+		logger.Info("Failed to delete backup from s3", "backup path", cr.Status.Destination, "error", err.Error())
+	} else {
+		logger.Info("backup was removed from s3", "name", cr.Name)
+	}
 
 	err = r.client.Update(context.TODO(), cr)
 	if err != nil {
 		logger.Error(err, "failed to update finalizers for backup", "backup", cr.Name)
 	}
 }
 
-func (r *ReconcilePerconaXtraDBClusterBackup) removeBackup(bucket, backup string, s3cli *minio.Client) error {
-	objs := s3cli.ListObjects(context.Background(), bucket,
-		minio.ListObjectsOptions{
-			UseV1:     true,
-			Recursive: true,
-			Prefix:    backup,
-		})
-
-	for v := range objs {
-		if v.Err != nil {
-			return errors.Wrap(v.Err, "failed to list objects")
-		}
+func removeBackup(bucket, backup string, s3cli *minio.Client) func() error {
+	return func() error {
+		objs := s3cli.ListObjects(context.Background(), bucket,
+			minio.ListObjectsOptions{
+				UseV1:     true,
+				Recursive: true,
+				Prefix:    backup,
+			})
```

Comment:
is it possible to remove all object which matches the pattern, without List?
is it possible with V2 version of the S3 protocol?

Reply:
1. As i can see no. We need to list objects and provide full info (idk why it accepts only this channel, instead of strings) plus s3 requires it. See - https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html.
2. 18b05213